### ISSUE TITLE
IDE0004 being reported on casting nint* to nint via void*

### ISSUE THREAD

**Issue by AlFasGD** [dotnet/roslyn#56207]
**Version Used**: VS 2019 Community 16.11.2 

**Steps to Reproduce**:

In the example code below, the cast is necessary to cast a dereferenced `nint*` into `nint`:
```csharp
nint** ptr = GetDoublePointer();
nint value = (nint)(void*)*ptr;
```

**Expected Behavior**: No cast redundancy warning.

**Actual Behavior**: IDE0004 is reported.

**Comment by Youssef1313** [comment]
Related to https://github.com/dotnet/roslyn/issues/50043.
